reportUploadTask/utils.ts

```diff
@@ -49,22 +48,6 @@ export function isCoverageFile(file: string): boolean {
 	return path.extname(file).toLowerCase() === '.coverage';
 }
 
-export function createUploadUrl(
-	teamscaleUrl: string,
-	project: string,
-	format: string,
-	partition: string,
-	message: string,
-	revision: string
-): string {
-	if (!teamscaleUrl.endsWith('/')) {
-		teamscaleUrl += '/';
-	}
-
-	const url = urlLib.parse(`${teamscaleUrl}p/${project}/external-report`, true);
-	url.query['format'] = format;
-	url.query['revision'] = revision;
-	url.query['partition'] = partition;
-	url.query['message'] = message;
-	return urlLib.format(url);
+export function isNullOrEmpty(s: string) {
```

Comment:
the checks you replaced were often not specifically for null, but also for `undefined`. Maybe that doesn't matter, but I'm not 100% sure. Please you decide if the current version is sufficient or if we should also be checking for undefined via

```js
return !s || s === '';
```

(this is what some of the old code did)

Reply:
AFAIK this does check for both `null` and `undefined` (it's `==`, not `===`).

Reply:
ah I see, I didn't catch that it's `s == null`. I think that's a bit too "tricky" for my taste. I'd prefer being explicit. If we want to check for undefined, do so explicitly with `===`. If not, then only check for null explicitly with `===`. It's too easy to overlook the `===` vs `==`, especially since Teamscale forces us to use `===` in most cases, so that's what I'm used to. I'm also a bit afraid that someone will try and "fix" this line by changing `==` to `===` and then break something

Reply:
Ok, I see. I also changed the name of the method to not suggest that it does explicitly check for _only_ `null`.
Usually, I only want to check whether the string is empty but still make it `null` and `undefined` safe.